### PR TITLE
test_errata fix api and cli

### PR DESCRIPTION
**Pull request by akhil-jha**
1. Fix for API and CLI tests. Mostly changing the job invocation task labels and searchquery.
2. Removed 1 API test as the corresponding BZ is moved to CLOSED WONTFIX

### CODE REVIEW

File: tests/foreman/api/test_errata.py [hunk 1]
```diff
@@ -204,24 +204,21 @@ def test_positive_install_in_hc(module_org, activation_key, custom_repo, default
...
-                'organization-id': f'{module_org.id}',
-            }
+                'inputs': {'errata': f'{CUSTOM_REPO_ERRATA_ID}'},
+                'targeting_type': 'static_query',
+                'search_query': f'host_collection_id = {host_collection.id}',
+                'organization_id': module_org.id,
+            },
+        )['id']
+        wait_for_tasks(
+            search_query=('label = Actions::RemoteExecution::RunHostsJob' f' and id = {task_id}'),
```
Comment (mshriver) [inline tests/foreman/api/test_errata.py:217]: Formatting of this string is screwy, `Black` likely formatted after you wrote the lines. 

Put the `f` in front of the entire string, and make it a single string.  (Right now its two strings)

**Review by mshriver — APPROVED** [review]

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

File: tests/foreman/api/test_errata.py [hunk 2]
```diff
@@ -204,24 +204,21 @@ def test_positive_install_in_hc(module_org, activation_key, custom_repo, default
...
-        JobInvocation.create(
-            {
+        task_id = default_sat.api.JobInvocation().run(
+            data={
                 'feature': 'katello_errata_install',
-                'inputs': f'errata=\'{CUSTOM_REPO_ERRATA_ID}\'',
-                'search-query': host_collection_query,
-                'organization-id': f'{module_org.id}',
-            }
+                'inputs': {'errata': f'{CUSTOM_REPO_ERRATA_ID}'},
```
Comment (JacobCallahan) [inline tests/foreman/api/test_errata.py:210]: Why f-string instead of str()

File: tests/foreman/cli/test_errata.py [hunk 3]
```diff
@@ -1382,20 +1384,27 @@ def test_apply_errata_using_default_content_view(errata_host):
...
-    # note time for later wait_for_tasks include 2 mins margin of safety.
-    timestamp = (datetime.utcnow() - timedelta(minutes=2)).strftime(TIMESTAMP_FMT)
     # Update errata from Library, i.e. Default CV
-    errata_host.run(f'yum -y update --advisory {REAL_0_ERRATA_ID}')
-    # Wait for upload profile event (in case Satellite system slow)
+    result = JobInvocation.create(
+        {
+            'feature': 'katello_errata_install',
+            'search-query': f'name = {errata_host.hostname}',
+            'inputs': f'errata=\'{REAL_0_ERRATA_ID}\'',
```
Comment (JacobCallahan) [inline tests/foreman/cli/test_errata.py:1392]: Use double quotes instead of string escapes

**Review by JacobCallahan — CHANGES_REQUESTED** [review]

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

File: tests/foreman/cli/test_errata.py (hunk 3 above)
Comment (akhil-jha) [reply tests/foreman/cli/test_errata.py:1392]: No can do. 
The `cmd` in `ssh.command()` gets constructed like this:
```
'LANG=en_US.UTF-8  hammer -v -u admin -p changeme --output=csv job-invocation create --feature="katello_errata_install" --search-query="name = <host>" --inputs="errata="RHBA-2021:1314"" --organization-id="4" '
```
^^The `inputs` arg gets messed up.

File: tests/foreman/api/test_errata.py (hunk 2 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py:210]: I don't follow this. Could you maybe rephrase the comment maybe?

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/cli/test_errata.py

File: tests/foreman/cli/test_errata.py (hunk 3 above)
Comment (akhil-jha) [reply tests/foreman/cli/test_errata.py]: Yes can do.

File: tests/foreman/api/test_errata.py (hunk 2 above)
Comment (akhil-jha) [reply tests/foreman/api/test_errata.py]: yep. Done

**Comment by akhil-jha** [comment]
trigger: test-robottelo
pytest: --importance High tests/foreman/api/test_errata.py

**Review by JacobCallahan — APPROVED** [review]
